avocado/core/nrunner.py

```diff
@@ -307,11 +310,57 @@ def run(self):
         yield queue.get()
 
 
+class TAPRunner(BaseRunner):
+    """Runner for standalone executables treated as TAP."""
+    def run(self):
+        env = self.runnable.kwargs or None
+        process = subprocess.Popen(
+            [self.runnable.uri] + list(self.runnable.args),
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            env=env)
+
+        yield {'status': 'started',
+               'timestamp': time.time()}
+
+        while process.poll() is None:
+            time.sleep(RUNNER_RUN_STATUS_INTERVAL)
+            yield {'status': 'running',
+                   'timestamp': time.time()}
+
+        stdout = io.TextIOWrapper(process.stdout)
+        parser = TapParser(stdout)
+        aborted = False
+        for event in parser.parse():
+            if isinstance(event, TapParser.Bailout):
+                aborted = True
+                break
+            elif isinstance(event, TapParser.Error):
+                result = 'error'
+                break
+            elif isinstance(event, TapParser.Test):
+                if event.result in (TestResult.XPASS, TestResult.FAIL):
+                    result = 'fail'
+                    break
+                elif event.result == TestResult.SKIP:
+                    result = 'skip'
+                    break
+                else:
+                    result = 'pass'
+
+        yield {'status': 'aborted' if aborted else 'finished',
```

Comment:
While I agree with removing the ambiguity of `status` and introducing `result`, we should have that resolved before introducing runners that produce incompatible output.  What I mean is that this runner, AS IS, can *not* be hooked into the existing `avocado nrun ...` workflow.

To do so, I had to:

```diff
-        yield {'status': 'started',
-               'timestamp': time.time()}
+        #yield {'status': 'started',
+        #       'timestamp': time.time()}
```

and:

```diff
-        yield {'status': 'aborted' if aborted else 'finished',
+        yield {'status': result, #'aborted' if aborted else 'finished',
```

Reply:
Hum, I see. 
When you revised and tested I tought that this would be ok, sorry. I will fix that.